### PR TITLE
Rewrite contributing guidelines to proper content for this repo

### PR DESCRIPTION
**Pull request by kulmann**
This PR updates the contributing guidelines to have proper content that matches this repo.

Partly fixes https://github.com/owncloud/web/issues/4666 (at least for this repo)
Fixes https://github.com/owncloud/web/issues/4655

### CODE REVIEW

**Comment by kulmann** [comment]
@pascalwengerter could you have a look if this makes more sense for you? The linked documentation doesn't cover all the required bits and pieces - we're still working on that. But that content will eventually appear on the linked documentation pages.

**Comment by kulmann** [comment]
cc @micbar @dragotin

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,50 +1,51 @@
...
 
 Thanks for wanting to contribute source code to ownCloud. That's great!
 
-Before we're able to merge your code into the ownCloud core, you need to sign our [Contributor Agreement][agreement]. You can sign it digitally as soon as you open your first pull request.
+Before we're able to merge your code, you need to sign our [Contributor Agreement][agreement]. You can sign it digitally as soon as you open your first pull request.
 
-Please read the [Developer Manuals][devmanual] to learn how to create your first application or how to test the ownCloud code with PHPUnit.
+Please read the [Developer Documentation][devmanual] to learn how to use our extension system or how to run tests for your features.
 
 In order to constantly increase the quality of our software we can no longer accept pull request which submit un-tested code.
```
Comment (phil-davis) [inline .github/CONTRIBUTING.md:41]:
```suggestion
In order to constantly increase the quality of our software we can no longer accept pull requests which submit un-tested code.
```

**Review by phil-davis — COMMENTED** [review]
LGTM - may as well fix this one tiny piece of grammar also :)

**Review by micbar — APPROVED** [review]
LGTM

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (micbar) [reply .github/CONTRIBUTING.md:41]: 👍

**Comment by pascalwengerter** [comment]
> @pascalwengerter could you have a look if this makes more sense for you? The linked documentation doesn't cover all the required bits and pieces - we're still working on that. But that content will eventually appear on the linked documentation pages.

Looks pretty neat to me now :)

**Pull request merged**